### PR TITLE
Dex,oauth2-proxy and k8s token documentation

### PR DESCRIPTION
**Pull request by juliusvonkohout**
@tarekabouzeid please review

supersedes https://github.com/kubeflow/manifests/pull/2888 @choudhryfrompak

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -228,22 +228,42 @@ echo "Installing oauth2-proxy..."
...
 
-# OPTION 1: works on most clusters, does NOT allow K8s service account 
+# OPTION 1: works on most clusters, does NOT allow K8s service account
 #           tokens to be used from outside the cluster via the Istio ingress-gateway.
 #
 kustomize build common/oauth2-proxy/overlays/m2m-dex-only/ | kubectl apply -f -
 kubectl wait --for=condition=ready pod -l 'app.kubernetes.io/name=oauth2-proxy' --timeout=180s -n oauth2-proxy
 
-# Option 2: works on Kind/K3D and other clusters with the proper configuration, and allows K8s service account tokens to be used
+# Option 2: works on Kind, K3D, Rancher, GKE and many other clusters with the proper configuration, and allows K8s service account tokens to be used
```
Comment (tarekabouzeid) [inline README.md:237]: RKE 1 doesn't support `.well-known/openid-configuration` . So option 1 works in rancher

Comment (juliusvonkohout) [reply README.md:237]: I am using it on rancher as well, so it definitely works :-) did you enable access in rancher to the openid configuration? SOmethign like this i guess https://github.com/rancher/rancher/issues/36444

**Pull request merged**